More tests, cleaning

tomasznapieralski opened the pull request:
Please CR:
@rochan 
@RodriGomez

tomasznapieralski commented:
Please CR:
@rochan 
@RodriGomez

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/LightBoxMercuryComponentObject.java:
```diff
@@ -73,5 +94,172 @@ public void verifyImageWasChanged(String imageOnePath, String imageTwoPath) {
+			direction = "left";
+		}
+		for (int i = 0; i < attempts; ++i) {
+			touchAction.swipeFromPointToPoint(startX, 50, endX, 50, duration, waitAfter);
```
Suggest moving the "50" to the top and use a variable/constant instead.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		}
+		try {
+			mobileDriver.swipe(startX, startY, endX, endY, duration);
+		} catch (Exception e) {}
```
Please use a more specific Exception that the try block is catching.
Also, please either throw the exception or log the exception in the catch block.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		} catch (Exception e) {}
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {}
```
Please use a more specific Exception that the try block is catching.
Also, please either throw the exception or log the exception in the catch block.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {}
+		if (mobileDriver.getContext() != "WEBVIEW_1") {
```
if doing a String comparison, it needs to be used with .equals()
Right now, this comparison would always hold true and trigger the next statement.
```
!"WEBVIEW_!".equals(mobileDriver.getContext())
```

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/MercuryArticlePageObject.java:
```diff
@@ -155,4 +155,5 @@ public void verifyCanonicalUrl() {
 	public void verifyRepliesAreExpanded() {
 		Assertion.assertTrue(checkIfElementOnPage(repliesContent.get(0)));
 	}
+	
```
remove blank line

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/MercuryBasePageObject.java:
```diff
@@ -36,6 +36,8 @@ public void openMercuryWiki(String wikiName){
 	public void tapOnElement(WebElement element) {
 		JavascriptExecutor jsexec = (JavascriptExecutor)driver;
 		jsexec.executeScript("arguments[0].click();", element);
+//		Actions flick = new Actions(driver);
```
Please either remove the line or put a comment on why this is commented out.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/MercuryBasePageObject.java:
```diff
@@ -36,6 +36,8 @@ public void openMercuryWiki(String wikiName){
 		JavascriptExecutor jsexec = (JavascriptExecutor)driver;
 		jsexec.executeScript("arguments[0].click();", element);
+//		Actions flick = new Actions(driver);
+//		flick.click(element).perform();
```
Please either remove the line or put a comment on why this is commented out.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		mobileDriver = NewDriverProvider.getMobileDriver();
+		driver = webDriver;
+		JavascriptExecutor js = (JavascriptExecutor) driver;
+		if (mobileDriver.getContext() != "NATIVE_APP") {
```
if doing a String comparison, it needs to be used with .equals()
Right now, this comparison would always hold true and trigger the next statement.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		try {
+			Thread.sleep(duration * 3);
+		} catch(Exception e) {}
+		if (mobileDriver.getContext() != "WEBVIEW_1") {
```
if doing a String comparison, it needs to be used with .equals()
Right now, this comparison would always hold true and trigger the next statement.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		int centerX = appNativeWidth / 2;
+		int centerY = (appNativeHeight / 2) + taskbarNativeHeight;
+		int path = 0;
+		if (mobileDriver.getContext() != "NATIVE_APP") {
```
if doing a String comparison, it needs to be used with .equals()
Right now, this comparison would always hold true and trigger the next statement.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {}
+		if (mobileDriver.getContext() != "WEBVIEW_1") {
```
if doing a String comparison, it needs to be used with .equals()
Right now, this comparison would always hold true and trigger the next statement.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		startY = (int)(((startY / 100f) * appNativeHeight) + taskbarNativeHeight);
+		endX = (int)((endX / 100f) * appNativeWidth);
+		endY = (int)(((endY / 100f) * appNativeHeight) + taskbarNativeHeight);
+		if (mobileDriver.getContext() != "NATIVE_APP") {
```
if doing a String comparison, it needs to be used with .equals()
Right now, this comparison would always hold true and trigger the next statement.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+			startXTwo = endXTwo;
+			endXTwo = temp;
+		}
+		if (mobileDriver.getContext() != "NATIVE_APP") {
```
if doing a String comparison, it needs to be used with .equals()
Right now, this comparison would always hold true and trigger the next statement.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {}
+		if (mobileDriver.getContext() != "WEBVIEW_1") {
```
if doing a String comparison, it needs to be used with .equals()
Right now, this comparison would always hold true and trigger the next statement.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+	public void tapOnPointXY(int pointX, int pointY, int duration, int waitAfter) {
+		pointX = (int)((pointX / 100f) * appNativeWidth);
+		pointY = (int)(((pointY / 100f) * appNativeHeight) + taskbarNativeHeight);
+		if (mobileDriver.getContext() != "NATIVE_APP") {
```
if doing a String comparison, it needs to be used with .equals()
Right now, this comparison would always hold true and trigger the next statement.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {}
+		if (mobileDriver.getContext() != "WEBVIEW_1") {
```
if doing a String comparison, it needs to be used with .equals()
Right now, this comparison would always hold true and trigger the next statement.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		js.executeScript("window.scrollTo(0, "+element.getLocation().getY()+")");
+		int pointX = (elementStartPointX + (elementWidth / 2)) * ratio;
+		int pointY = (taskbarWebviewHeight + (elementHeight / 2) + offSetY) * ratio; 
+		if (mobileDriver.getContext() != "NATIVE_APP") {
```
if doing a String comparison, it needs to be used with .equals()
Right now, this comparison would always hold true and trigger the next statement.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {}
+		if (mobileDriver.getContext() != "NATIVE_APP") {
```
if doing a String comparison, it needs to be used with .equals()
Right now, this comparison would always hold true and trigger the next statement.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		int duration = startY - endY;
+		try {
+			mobileDriver.swipe(startX, startY, endX, endY, duration);
+		} catch (Exception e) {}
```
Please use a more specific Exception that the try block is catching.
Also, please either throw the exception or log the exception in the catch block.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		} catch (Exception e) {}
+		try {
+			Thread.sleep(duration * 3);
+		} catch(Exception e) {}
```
Please use a more specific Exception that the try block is catching.
Also, please either throw the exception or log the exception in the catch block.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+				}
+				try {
+					mobileDriver.swipe(centerX, centerY, centerX - path, centerY, duration);
+				} catch (Exception e) {}
```
Please use a more specific Exception that the try block is catching.
Also, please either throw the exception or log the exception in the catch block.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+				}
+				try {
+					mobileDriver.swipe(centerX, centerY, centerX + path, centerY, duration);	
+				} catch (Exception e) {}
```
Please use a more specific Exception that the try block is catching.
Also, please either throw the exception or log the exception in the catch block.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+				}
+				try {
+					mobileDriver.swipe(centerX, centerY, centerX, centerY - path, duration);				
+				} catch (Exception e) {}
```
Please use a more specific Exception that the try block is catching.
Also, please either throw the exception or log the exception in the catch block.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+				}
+				try {						
+					mobileDriver.swipe(centerX, centerY, centerX, centerY + path, duration);	
+				} catch (Exception e) {}
```
Please use a more specific Exception that the try block is catching.
Also, please either throw the exception or log the exception in the catch block.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		}
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {}
```
Please use a more specific Exception that the try block is catching.
Also, please either throw the exception or log the exception in the catch block.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		multiTouch.add(touchTwo);
+		try {
+			multiTouch.perform();
+		} catch (Exception e) {}
```
Please use a more specific Exception that the try block is catching.
Also, please either throw the exception or log the exception in the catch block.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		} catch (Exception e) {}
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {}
```
Please use a more specific Exception that the try block is catching.
Also, please either throw the exception or log the exception in the catch block.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		}
+		try {
+			mobileDriver.tap(1, pointX, pointY, duration);
+		} catch (Exception e) {}
```
Please use a more specific Exception that the try block is catching.
Also, please either throw the exception or log the exception in the catch block.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		} catch (Exception e) {}
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {}
```
Please use a more specific Exception that the try block is catching.
Also, please either throw the exception or log the exception in the catch block.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		}
+		try {
+			mobileDriver.tap(1, pointX, pointY, duration);
+		} catch (Exception e) {}
```
Please use a more specific Exception that the try block is catching.
Also, please either throw the exception or log the exception in the catch block.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		} catch (Exception e) {}
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {}
```
Please use a more specific Exception that the try block is catching.
Also, please either throw the exception or log the exception in the catch block.

rochan commented on src/test/java/com/wikia/webdriver/testcases/mercurytests/InteractiveMapsTests.java:
```diff
@@ -22,75 +22,83 @@ public void optInMercury() {
-		maps.verifyPinPopUpAppeared();
+		maps.verifyMapIdInUrl();
 	}
+
```
remove blank line

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/LightBoxMercuryComponentObject.java:
```diff
@@ -73,5 +94,172 @@ public void verifyImageWasChanged(String imageOnePath, String imageTwoPath) {
+		int endX = 20;
+		int duration = 300;
+		int waitAfter = 5000;
+		if (direction == "right") {
```
if doing a String comparison, it needs to be used with .equals()
I would suggest putting the String on the left side to avoid NullPointerException
```
"right".equals(direction)
```

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/LightBoxMercuryComponentObject.java:
```diff
@@ -73,5 +94,172 @@ public void verifyImageWasChanged(String imageOnePath, String imageTwoPath) {
+			startX = endX;
+			endX = temp;
+		} else {
+			direction = "left";
```
if doing a String comparison, it needs to be used with .equals()
I would suggest putting the String on the left side to avoid NullPointerException

tomasznapieralski replied:
Changed:
if (direction == "right") {...}
To:
if ("right".equals(direction)) {...}

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/LightBoxMercuryComponentObject.java:
```diff
@@ -73,5 +94,172 @@ public void verifyImageWasChanged(String imageOnePath, String imageTwoPath) {
+		int waitAfter = 5000;
+		boolean lastDisplayVisibility = true;
+		if (LightboxHeaderDisplayValue.contains(LightboxFooterDisplayValue)) {
+			if (LightboxHeaderDisplayValue == "none") {
```
if doing a String comparison, it needs to be used with .equals()
I would suggest putting the String on the left side to avoid NullPointerException

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/LightBoxMercuryComponentObject.java:
```diff
@@ -73,5 +94,172 @@ public void verifyImageWasChanged(String imageOnePath, String imageTwoPath) {
+		int pointX = 25;
+		int duration = 500;
+		int waitAfter = 5000;
+		if (edge == "right") {
```
if doing a String comparison, it needs to be used with .equals()
I would suggest putting the String on the left side to avoid NullPointerException

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/LightBoxMercuryComponentObject.java:
```diff
@@ -73,5 +94,172 @@ public void verifyImageWasChanged(String imageOnePath, String imageTwoPath) {
+		if (edge == "right") {
+			pointX = 75;
+		} else {
+			edge = "left";
```
if doing a String comparison, it needs to be used with .equals()
I would suggest putting the String on the left side to avoid NullPointerException

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+		if (startXTwo >= (appNativeWidth - 2)) {
+			startXTwo = appNativeWidth - 3;
+		}
+		if (zoomWay == "out") {
```
if doing a String comparison, it needs to be used with .equals()
I would suggest putting the String on the left side to avoid NullPointerException

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+	/**
+	 * That method return value of all resolution variables.
+	 */
+	public void varStatus() {
```
Is there any other output method that you could use besides System.out ?

tomasznapieralski replied:
We will use varStatus() method only if we will test our tests :) I forgot to remove it from constructor.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/LightBoxMercuryComponentObject.java:
```diff
@@ -73,5 +94,172 @@ public void verifyImageWasChanged(String imageOnePath, String imageTwoPath) {
+		} else {
+			lightboxFooterDisplay = true;
+		}
+		if (lightboxHeaderDisplay == lastDisplayVisibility) {
```
contains() returns a boolean, so you could use:
```
lightboxHeader.getCssValue("display").contains("none");
```

rochan replied:
Then you don't need any of the lines above

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/LightBoxMercuryComponentObject.java:
```diff
@@ -73,5 +94,172 @@ public void verifyImageWasChanged(String imageOnePath, String imageTwoPath) {
+		} else {
+			PageObjectLogging.log("verifyChangesInUI", "Lightbox header visibility changed", true);
+		}
+		if (lightboxFooterDisplay == lastDisplayVisibility) {
```
contains() returns a boolean, so you could use:
```
lightboxFooter.getCssValue("display").contains("none");
```

tomasznapieralski replied:
Applied

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/LightBoxMercuryComponentObject.java:
```diff
@@ -73,5 +94,172 @@ public void verifyImageWasChanged(String imageOnePath, String imageTwoPath) {
+				break;
+		}
+		File afterZoomIn = shooter.capturePage(driver);
+		if (!ic.areFilesTheSame(beforeZooming, afterZoomIn)) {
```
Please switch the if else and avoid using negation '!' for better readability

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/LightBoxMercuryComponentObject.java:
```diff
@@ -73,5 +94,172 @@ public void verifyImageWasChanged(String imageOnePath, String imageTwoPath) {
+		touchAction.tapOnPointXY(50, 50, 140, 0);
+		touchAction.tapOnPointXY(50, 50, 140, 2000);
+		File afterZooming = shooter.capturePage(driver);
+		if (!ic.areFilesTheSame(beforeZooming, afterZooming)) {
```
Please switch the if else and avoid using negation '!' for better readability

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/LightBoxMercuryComponentObject.java:
```diff
@@ -73,5 +94,172 @@ public void verifyImageWasChanged(String imageOnePath, String imageTwoPath) {
+		}
+		touchAction.swipeFromCenterToDirection(direction, 200, 200, 2000);
+		File afterMoving = shooter.capturePage(driver);
+		if (!ic.areFilesTheSame(afterZooming, afterMoving)) {
```
Please switch the if else and avoid using negation '!' for better readability

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/componentobject/mercury/SmartBannerComponentObject.java:
```diff
@@ -45,16 +54,46 @@ public void clickInstallButton() {
+		int firstPosition = smartBanner.getLocation().getY();
+		touchAction.swipeFromPointToPoint(50, 90, 50, 40, 500, 3000);
+		int secondPosition = smartBanner.getLocation().getY();
+		if (firstPosition == secondPosition) {
```
No need for variables if they are only being used once.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java:
```diff
@@ -766,4 +767,28 @@ public EmbedMapComponentObject clickViewEmbedMap() {
+	 */
+	public static int[] convertRGBAFunctiontoIntTable(String rgba) {	
+		int[] extract = new int[3];
+		int start = 0;
```
no need to initialize it to 0 since the assignment is right after.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/article/ArticlePageObject.java:
```diff
@@ -766,4 +767,28 @@ public EmbedMapComponentObject clickViewEmbedMap() {
+	public static int[] convertRGBAFunctiontoIntTable(String rgba) {	
+		int[] extract = new int[3];
+		int start = 0;
+		int end = 0;
```
no need to initialize it to 0 since the assignment is right after.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,360 @@
+	private static int appWebviewWidth = 0;
+	
+	private static int loadedPageHeight = 0;
+	private static int loadedPageWidth = 0;
```
What is the reason that this variable needs to be static?
Normally, we want to definite it with 'final' to be 'static final'.

tomasznapieralski commented:
Thanks @rochan for CR ;)
Can we merge it now?

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		int duration = startY - endY;
+		try {
+			mobileDriver.swipe(startX, startY, endX, endY, duration);
+		} catch (Exception e) {
```
Please specified which exception the try block is catching rather than a generic exception.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		}
+		try {
+			Thread.sleep(duration * 3);
+		} catch(Exception e) {
```
Please specified which exception the try block is catching rather than a generic exception.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+				}
+				try {
+					mobileDriver.swipe(centerX, centerY, centerX - path, centerY, duration);
+				} catch (Exception e) {
```
Please specified which exception the try block is catching rather than a generic exception.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+				}
+				try {
+					mobileDriver.swipe(centerX, centerY, centerX + path, centerY, duration);	
+				} catch (Exception e) {
```
Please specified which exception the try block is catching rather than a generic exception.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+				}
+				try {
+					mobileDriver.swipe(centerX, centerY, centerX, centerY - path, duration);				
+				} catch (Exception e) {
```
Please specified which exception the try block is catching rather than a generic exception.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+				}
+				try {						
+					mobileDriver.swipe(centerX, centerY, centerX, centerY + path, duration);	
+				} catch (Exception e) {
```
Please specified which exception the try block is catching rather than a generic exception.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		}
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {
```
Please specified which exception the try block is catching rather than a generic exception.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		}
+		try {
+			mobileDriver.swipe(startX, startY, endX, endY, duration);
+		} catch (Exception e) {
```
Please specified which exception the try block is catching rather than a generic exception.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		}
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {
```
Please specified which exception the try block is catching rather than a generic exception.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		multiTouch.add(touchTwo);
+		try {
+			multiTouch.perform();
+		} catch (Exception e) {
```
Please specified which exception the try block is catching rather than a generic exception.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		}
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {
```
Please specified which exception the try block is catching rather than a generic exception.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		}
+		try {
+			mobileDriver.tap(1, pointX, pointY, duration);
+		} catch (Exception e) {
```
Please specified which exception the try block is catching rather than a generic exception.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		}
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {
```
Please specified which exception the try block is catching rather than a generic exception.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		}
+		try {
+			mobileDriver.tap(1, pointX, pointY, duration);
+		} catch (Exception e) {
```
Please specified which exception the try block is catching rather than a generic exception.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		}
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {
```
Please specified which exception the try block is catching rather than a generic exception.

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {
+			PageObjectLogging.log("tapOnPointXY", e.toString(), false);
```
Use e.getMessage()

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		try {
+			mobileDriver.swipe(startX, startY, endX, endY, duration);
+		} catch (Exception e) {
+			PageObjectLogging.log("PerformTouchAction", e.toString(), false);
```
Use e.getMessage()

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		try {
+			Thread.sleep(duration * 3);
+		} catch(Exception e) {
+			PageObjectLogging.log("PerformTouchAction", e.toString(), false);
```
Use e.getMessage()

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+				try {
+					mobileDriver.swipe(centerX, centerY, centerX - path, centerY, duration);
+				} catch (Exception e) {
+					PageObjectLogging.log("swipeFromCenterToDirection", e.toString(), false);
```
Use e.getMessage()

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+				try {
+					mobileDriver.swipe(centerX, centerY, centerX + path, centerY, duration);	
+				} catch (Exception e) {
+					PageObjectLogging.log("swipeFromCenterToDirection", e.toString(), false);
```
Use e.getMessage()

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+				try {
+					mobileDriver.swipe(centerX, centerY, centerX, centerY - path, duration);				
+				} catch (Exception e) {
+					PageObjectLogging.log("swipeFromCenterToDirection", e.toString(), false);
```
Use e.getMessage()

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+				try {						
+					mobileDriver.swipe(centerX, centerY, centerX, centerY + path, duration);	
+				} catch (Exception e) {
+					PageObjectLogging.log("swipeFromCenterToDirection", e.toString(), false);
```
Use e.getMessage()

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {
+			PageObjectLogging.log("swipeFromCenterToDirection", e.toString(), false);
```
Use e.getMessage()

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		try {
+			mobileDriver.swipe(startX, startY, endX, endY, duration);
+		} catch (Exception e) {
+			PageObjectLogging.log("swipeFromPointToPoint", e.toString(), false);
```
Use e.getMessage()

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {
+			PageObjectLogging.log("swipeFromPointToPoint", e.toString(), false);
```
Use e.getMessage()

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		try {
+			multiTouch.perform();
+		} catch (Exception e) {
+			PageObjectLogging.log("zoomInOutPointXY", e.toString(), false);
```
Use e.getMessage()

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {
+			PageObjectLogging.log("zoomInOutPointXY", e.toString(), false);
```
Use e.getMessage()

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		try {
+			mobileDriver.tap(1, pointX, pointY, duration);
+		} catch (Exception e) {
+			PageObjectLogging.log("tapOnPointXY", e.toString(), false);
```
Use e.getMessage()

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		try {
+			Thread.sleep(waitAfter);
+		} catch (Exception e) {
+			PageObjectLogging.log("tapOnPointXY", e.toString(), false);
```
Use e.getMessage()

rochan commented on src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/mercury/PerformTouchAction.java:
```diff
@@ -0,0 +1,390 @@
+		try {
+			mobileDriver.tap(1, pointX, pointY, duration);
+		} catch (Exception e) {
+			PageObjectLogging.log("tapOnPointXY", e.toString(), false);
```
Use e.getMessage()

rochan commented:
@tomasznapieralski There are noticeable amounts of try and catch block on a few selected methods. It would be a good idea to create a private method as a wrapper that would handle the try/catch to reduce the same block of code.

tomasznapieralski commented:
Fixed.
As you suggested @rochan i put try-catch in separate methods.
Also i used InterruptedException for Thread.sleep() and NoSuchContextException for AppiumDriver.context() instead of Exception.
I removed handling for swipe, tap etc. because they didn't throw exception. When i got error it was because of wrong context which is now handled.

rochan commented:
LGTM. Good job.
